processor/src/it/java/com/linecorp/decaton/processor/CoreFunctionalityTest.java

```diff
@@ -103,4 +114,49 @@ public void testAsyncTaskCompletion() {
                 .build()
                 .run();
     }
+
+    /**
+     * Decaton doesn't guarantee that produced messages are processed in
+     * "strictly" same order (refs {@link GuaranteeType#PROCESS_ORDERING}).
+     *
+     * But as long as we set partition.concurrency to 1 and all subscriptions are
+     * restarted gracefully, we can expect "strict" process ordering as like vanilla Kafka consumer
+     */
+    @Test(timeout = 60000)
+    public void testSingleThreadProcessing() {
+        ProcessingGuarantee strictOrdering = new ProcessingGuarantee() {
```

Comment:
Is strictOrdering` what we expect for this? I thought it's something more a like "no duplicates"?

Reply:
thanks, `noDuplicates` sounds more appropriate.